synapse/rest/media/v1/_base.py

```diff
@@ -49,18 +49,20 @@
 
 def parse_media_id(request: Request) -> Tuple[str, str, Optional[str]]:
     try:
+        # The type on postpath seems incorrect in Twisted.
```

Comment:
Maybe add the Twisted version to this comment, so it's easier to check if it's still relevant when we stumble across this comment in a year

Reply:
(+ if you got a issue # for all of those problems on `twisted/twisted`, maybe add then to the ones currently causing trouble in mypy, or a `fixme` at least)

Reply:
```suggestion
        # The type on postpath seems incorrect in Twisted 21.2.0.
```